scalasql/core/src/DbApi.scala

```diff
@@ -219,7 +219,26 @@ object DbApi {
         }
       }
     }
-
+  
+    def runOption[Q, R](query: Q, fetchSize: Int = -1, queryTimeoutSeconds: Int = -1)(
```

Comment:
Do we need this on `DbApi`. `.single` is implemented on `Select` and doesn't need any dedicated methods on `DbApi`, so I would expect `singleOption` would be similar